shared/js/background/events/3p-tracking-cookie-blocking.js

```diff
@@ -12,12 +12,16 @@ function dropTracking3pCookiesFromResponse (request) {
     const tab = tabManager.get({ tabId: request.tabId })
     let responseHeaders = request.responseHeaders
 
-    if (tab && tab.site.isFeatureEnabled('trackingCookies3p') && request.type !== 'main_frame') {
-        if (!trackerutils.isTracker(request.url)) {
+    if (tab && request.type !== 'main_frame') {
+        const requestIsTracker = trackerutils.isTracker(request.url)
+        if (requestIsTracker && !tab.site.isFeatureEnabled('trackingCookies3p')) {
+            return { responseHeaders }
+        }
+        if (!requestIsTracker && !tab.site.isFeatureEnabled('nonTracking3pCookies')) {
             return { responseHeaders }
         }
 
-        // Strip 3rd party response header
+        // Strip 3rd party cookie response header
         if (!request.responseHeaders) return { responseHeaders }
```

Comment:
Why don't we do this in the parent if statement? If we don't have the response/request headers these functions should abort asap.